### ISSUE TITLE
Feature Request: Disabling weapon attribute

### ISSUE THREAD

**Issue by Zitchas** [endless-sky/endless-sky#4394]
**Is your feature request related to a problem? Please describe.**
- Weapons (especially tail end of missile volleys) often kill ships, especially smaller ships.
- Projectiles seem quite similar and not much benefit to alternate missile types.
- Opportunities for "dueling" weapons (useful for EVN in ES plug-ins

**Related Issue Links**
People having problems with small ships being destroyed by the tail ends of volleys: https://github.com/endless-sky/endless-sky/issues/4149

**Describe the solution you'd like**
An attribute that can be specified for weapons that cause it to do 25% of normal hull damage when the target is disabled. 

**Additional context**
There are three main reasons for implementing this attribute: Lore, Missions, and early-game boarding. A fourth is for plug-ins. 

1) Despite the name "hull" damage, it seems broadly reasonable to assume that this damage includes things such as damage that focuses on crippling shipboard systems such as power, life support, fuel, engines, etc. This is particularly true for weapons that deal ionization, slowing, fuel destruction, or other effects.

As such, it seems reasonable to thing that such systems would be significantly less effective when the target ship isn't functioning. Electrical systems aren't going to suffer as much damage from ionization when they aren't running, etc.

2) On the flip side, the opportunity to have dueling weapons (as in weapons that are blunted, or otherwise have their ultra lethal capabilities reduced or re-calibrated so that they excel at stripping armour but minimize the damage done to life support and other critical systems) would open up a variety of interesting missions where the goal is to disable the other combatant in "fight to disable" style fights. On the flip side, it could be quite interesting to have opponents similarly equipped and set to disable, not destroy, so that the player isn't killed when defeated, but instead has the opportunity to be repaired and try again. While such fights are possible without disabling weapons, it would be possible to send the player up against much more challenging fights (including on par combats) without inducing unnecessary deaths or hardship to newer players.

3) One imbalance in the game is that the boarding and salvaging part of the game (particularly for those using missiles) is significantly handicapped in the early game when the player may be seriously depending on looting fighters, interceptors, and other small craft. The opportunity to buy weapons that are designed to target external armour and avoid critical systems would be a great boon to this early strategy.

Conversely, at the later stages of the game, most players are focused on heavy warships that have sufficient health that a few hits more or less are unlikely to have a significant impact on their ability to salvage from said ships. As such, weapons with these attributes would have little to no impact on late game capturing/salvaging, but would improve the early game capturing/salvaging.

The flip side to this is that pirates (who are by definition interested in looting the player) would want to be equipped with these weapons, which would significantly increase the survival rate of players in the early game, although potentially at the cost of their bank accounts. 

Having at least one primary weapon and one projectile weapon accessible to pirates in the early game could thus enable the early game combat to be more interesting but less lethal, and thus reduce some of the complaints about early game fatalities.

4) A EVN total conversion plug-in is in development, and that game included a number of weapons with the disabling property, primarily used for dueling weapons in the Auroran empire. Such an attribute would enable this project to stick much closer to the original game.

**Comment by ChamEV** [comment]
In addition to this, I think introducing a blanket damage reduction of ~75% on all weapons hitting a disabled ship for up to 2 seconds after it becomes disabled would be beneficial.

At the same time this is implemented, it should be accompanied by a re-balance of boarding and looting.  I'm not sure what such a system would look like, but having additional chances for the boarded ship to expire on looting, or loot to be too badly damaged to use.

**Comment by ghoulavenger** [comment]
If you just want to disable ships without destroying them, use beam weapons. Beam weapons stop firing as soon as a ship is disabled, allowing you to do whatever you want with them. So this problem in my opinion is already largely solved.

That said the larger problem with ships dying from my experience is crossfire. On smaller targets sure, most weapons might destroy rather than disable (except beam weapons which stop firing on disable). But on larger targets it's largely fleet size (aka overkill which beam weapons automatically correct) and crossfire (shooting through ships that are in your line of fire to other enemy ships). Even if the damage you deal with weapons is significantly lower, it doesn't change the fact that you're shooting through ships to target other ships, making these weapons actually piss poor in execution -- the crossfire kills are actually beneficial to defeating more opponents (since they are likely still shooting at your fleet), even if it means you get less loot. Getting rid of crossfire isn't really a reasonable solution either, since that means the battlefield will become cluttered very quickly and will also make capturing ships significantly easier.

So my feeling is that maybe it'll be most useful for the plugin you're talking about. But I couldn't see myself using any of this in regular play. Beam weapons are already superior to what you're suggesting and serve the same purpose (disable not kill) and crossfire isn't going anywhere.

Just for completionist sake I'll tackle your reasonings directly as well.

1. I grant you that this is a completely reasonable interpretation of the data. However, these weapons are already balanced to compensate for their inefficiency. If you want a weapon that is less lethal, you only need to adjust it some more.

2. Dueling weapons are an interesting idea, but without separating armor from hull kind of difficult to implement. So this change could be great if you were to change some other things too, such as the inclusion of armor or damage reduction. Lowering hull damage at certain thresholds could make be a reasonable facsimile but still won't outstrip a beam weapon that just stops at disabling the opponent.

3. Early boarding and salvaging believe it or not is best done with a starbarge without any weapons on it. Small ships are impractical for fighting, and just as likely to die, even using the best of missiles your fighter can equip. This is because you have no significant edge on your opponent, you're basically a flying tin can. 

Why the star barge? Because I can install a bunk on it to get to 7 crew, and it already has the most carrying capacity of the starter vehicles. Why no weapons? Well, if you really insist on boarding, you're going to be scumming kills from the navy probably. Even installing the best weapon a star barge can, you're not going to shoot down anything. You might reasonably try an AM turret, but that will limit your capturing potential further without significant gains in survivability.

4. Plugins. I can't argue with anything for plugins sake. Plugins are at the sole discretion of their creators.

**Comment by Zitchas** [comment]
I appreciate the details on better methods of disabling and salvaging, and I agree that even with the addition of the disabling attribute on certain weapons, beams will remain the best choice for doing so. That being said, having this attribute will allow certain projectile and missile weapons to regain some degree of usefulness in this regards. The intent isn't to replace beam's functionality in this regard, but to give both players and NPCs some additional options, especially in the early stages of the game.

1. There is no possible way to adjust the current weapons to reduce their lethality (as in reduce their tendency to destroy disabled ships) without reducing their effectiveness against hull in general combat. If certain weapons were given the `disabling` tag as described here, I don't believe that they would need any adjustments to their current stats.

2. This was precisely the intention: To simulate the complexity of having weapons that are optimized for something other than general-purpose-punching-holes-in-stuff. Whether that is "weapons that are primarily effective against operational systems" or "weapons that have been calibrated to target surface armor instead of internal systems", both are reasonably handled by the same tag and can simply be described as such in the descriptions and lore without needing further mechanic changes.  

3. I am reasonably well aware of the tactics for ideal ninja-salvaging. To add to your point, a completely unarmed ship will be de-prioritized as a target by hostile ships in the area, which makes it easier to perform salvaging operations. That being said, this change isn't intended to facilitate ninja-salvaging, but rather to open up alternative tools to enable the player to have a better chance at disabling ships in one-on-one combat against other small ships. Basically, to give players the chance to disable things while flying small ships without forcing them to use beams. 

More importantly, it allows us (content creators) to equip hostile ships in such a way that they can be effective in combat with missiles and longer range weaponry while still staying true to that ship's purpose as a pirate or a looter. Despite their label, many marauders and pirates who are supposedly focused on looting, robbing, or capturing ships are equipped in such a way that they have a poor chance of actually doing so, and instead destroying their targets. As a result, they seem more like berserk barbarians instead of pirates or looters.

4. More variety in plug-in weaponry. What more needs to be said?

*Implementation*

If this mechanic is implemented, I predict the following in-game modifications:

A) Certain in-game weapons would be given the `Disabling` tag. These weapons are those whose descriptions inherently sound like they are built to disable ship systems and thus could be expected to be less effective at simply bashing on inert metal: The Inhibitor Canon, Ion Canon, Ion Rain Canon, and EMP torpedo. 

B) Certain other weapons would have modified versions created Depending on the culture or company that produces the modified weapon, these would be described as either:
 - Weapons modified to increase their effectiveness at targeting ship operation systems such as energy distribution, ship command & control, hydraulic systems, etc. (most likely would have alternate version of the weapon available that are labeled "(disabling)"
 - Weapons modified to increase the damage dealt at the impact surface (aka outer hull) and minimize damage to internal systems (such as life support). If one takes a look at modern Armor Piercing ammunition, one can realize that there are countless ways one could modify ammunition in the opposite manner. Weapons modified in this way would probably be referred to as "(dueling)", although for simplicity they could be simply labeled "(disabling)" instead.
A variety of weapons in the game could have modified versions.  Most solid projectiles using missiles or bullet style projectiles could easily have suitable modified versions created: Meteors, Sidewinders, Javelins, Gatling gun, Railgun, and thunderheads. 

Weapons in the B category would likely be less common in outfitters than their standard versions, but should be available in a variety of places nonetheless. Pirate planets would also carry the variants of human weapons. In certain cultures (such as the Pirate, Remnant, and Coalition) modified/disabling versions of their weapons may be more common than the standard versions. (for the former, because they want to loot ships, and for the later due to the Heliarchs restriction on selling lethal weaponry to civilians)

**Comment by ghoulavenger** [comment]
>There is no possible way to adjust the current weapons to reduce their lethality (as in reduce their tendency to destroy disabled ships) without reducing their effectiveness against hull in general combat. If certain weapons were given the disabling tag as described here, I don't believe that they would need any adjustments to their current stats.

Yes, I was suggesting you reduce their effectiveness against hull period. It would pretty much accomplish the same thing. Because even if these weapons were to taper off their damage automatically as the ship became weaker, the net effect is still the same and completely indistinguishable to the player.
> That being said, this change isn't intended to facilitate ninja-salvaging, but rather to open up alternative tools to enable the player to have a better chance at disabling ships in one-on-one combat against other small ships.

Now see, here's the rub. Small ships are intended to be suicide wagons. By trying to make them stronger with like, missiles, you might possibly disable a stronger ship. But then you have the tradeoff of overkill on smaller targets. I believe this is the way it should be.

On your proposition to change weapons already in game that might fit your mold, I don't think they need to be changed. I'm more likely to accept that there is a modified variant. But even then, the only culture I know of that is interested in dueling specifically are the humans. The Remnant as human variant, might also have something specific to salvaging but seeing as their major opponents (The Korath Exiles) only use large ships is definitely going to be something uncommon. I don't see any other species really endorsing these weapons at all.

You mentioned the Coalition, but they don't want their civilians armed at all. Only the Heliarchs have weapons, and they're used explicitly for killing the Quarg. The only thing I might foreseeably see the Coalition armed with are mining weapons, which would probably already be weak enough without the proposed disabling tag.

**Comment by Zitchas** [comment]
Actually, it wouldn't be anywhere close to the same thing, because having a gradually decreasing effectiveness would make ships take longer to be disabled. This tag is intended to allow weapons act as per normal all the way up until the ship ceases to function, and then drop off sharply. The point is to have an attribute that doesn't affect the combat effectiveness of a weapon, while reducing its effectiveness against disabled targets. 

No, they're not intended to be suicide wagons. That would be fighters, and right now, it is physically impossible for players to fly anything classified as a fighter. (not to be confused with interceptors, which can be flown). Even if they were, though, this is not a good thing. There is already enough in the game forcing players to switch to the biggest possible ship available as soon as possible, why not give pirates (who actually don't want to kill things anyway) and players who want to enjoy the smaller game some options?

Anyway, those weapons I mentioned are options. Things that I think would be worth considering. Whether or not they were actually given it is up to the community and a future PR beyond this. I mentioned them just as examples. 

As for races that endorse them? Why do any need to endorse them? Likewise, you already said it yourself that you could see humans using these. And for most players, they'll never leave human space in the early game or in small ships. So the target audience for weapons with these abilities is primarily those flying (and dealing with) small human ships anyway. 

That aside, the Remnant absolutely would use these. They'd have non-disabling weapons, for sure. The thrasher canons and various other weapons currently in the pipeline will definitely be lethal. But for dealing with non-existential threats like raiding Korath that they lure in to pillage? Definitely a target for disabling weapons. 

The coalition I mentioned as a possibility. True, the Heliarchs don't want their civilians armed at all. That being said, it's been a *long* time since the Quarg were "driven out," and yet there is a massive armed Heliarch fleet everywhere in their space. They clearly have other problems to deal with that simply haven't been created yet, unless they happen to be the one group ever to actually maintain a generations long watch for a threat that no-one has seen. Once Coalition story is more fleshed out, I'm sure there's going to be reasons for so many armed ships, and with those reasons may come reasons for them to want to disable instead of kill. But who knows? That's all speculation.

**Comment by ghoulavenger** [comment]
> Actually, it wouldn't be anywhere close to the same thing, because having a gradually decreasing effectiveness would make ships take longer to be disabled. This tag is intended to allow weapons act as per normal all the way up until the ship ceases to function, and then drop off sharply. The point is to have an attribute that doesn't affect the combat effectiveness of a weapon, while reducing its effectiveness against disabled targets.

The reason why beam weapons are kind of a niche case is because of the time between the last shot (which usually happens when the exact moment the ship is disabled), and the last shot connecting (beams are instant, where no other weapons are). This means beams always stop at exactly the right moment, and all other weapons are still firing at you. 

Your proposal is to keep those weapons 100% effective except for those last rounds that hit after you disabled them. Well it might work a little bit better than a straight nerf, but it kind of flies in the face of the very idea you were originally proposing. If hull, armoring, and ship systems are the reason why these weapons hit so much harder, than a ship that is limping should take less damage across the board, not just at the moment of disabling. Perhaps though this is just a difference in opinion.

> That aside, the Remnant absolutely would use these. They'd have non-disabling weapons, for sure. The thrasher canons and various other weapons currently in the pipeline will definitely be lethal. But for dealing with non-existential threats like raiding Korath that they lure in to pillage? Definitely a target for disabling weapons.

And their thrashers would disable Korath raiders quite effectively, quite possibly even in large fleets, same with the worldships. The only question would be about the Chasers, and those are never used by themselves and definitely not worth salvaging if they can board a raider instead. Give me an enemy they might actually use them on, and I'll buy it, since they're salvagers. Basically they start producing it when they actually need it.

On the Coalition/Heliarch angle, right now I'm actually anticipating a civil war. The Heliarchs seem like an autocracy to me, which leaves open political unrest. In such case the Coalition (might) be willing to use weapons designed to allow boarding to help take some of the superior Heliarch technology into their fold (personally I'd think more weapons that prevented hyperspacing out by utilizing knockback). But like you said, this is, very speculatory at best, we simply don't have enough information.

**Comment by Zitchas** [comment]
I am quite well aware of how beams are useful for this. 

In the mechanics of how ES ship systems work, baring effects such as ionization or slowing, ships function at 100% until disabled. If, someday, we introduce a mechanic where ship performance degrades steadily as hull is damaged until they reach the point of being disabled, then yes, a "disabling" flagged weapon should also have a similar degradation in effect the closer a ship gets to being disabled. Until such a time as that is implemented, however, a disabling flagged weapon should follow the same performance drop off that ships use in relation to hull strength. That is to say, zero degradation until disabled, and then full degradation. 

Yes, I am quite well aware of how effective thrashers are. I spend a lot of my game time working out combat situations for/against the Remnant, for what are probably obvious reasons.

**Comment by ghoulavenger** [comment]
I wasn't trying to point out in my last post that beams are useful for that. I was trying to point out the problem with other weapons. This is key to the problem that you're posing. You want other weapons to be realistic options to beam weapons for disabling. The problem with other weapons is that they have no prediction on when a target is disabled. You're trying to correct the problem with a pragmatic solution rather than a direct one. A direct solution is to simply make other weapons stop firing sooner than beam weapons do. This is a very hard thing to do, because you would have to calculate every projectile flying at the ship in question, it's likelihood to hit, etc. Now perhaps this is feasible, I don't know, I'd need someone like tehhowch to tell me, but to me it seems like a lot of extra calculations.

> If, someday, we introduce a mechanic where ship performance degrades steadily as hull is damaged until they reach the point of being disabled, then yes, a "disabling" flagged weapon should also have a similar degradation in effect the closer a ship gets to being disabled. Until such a time as that is implemented, however, a disabling flagged weapon should follow the same performance drop off that ships use in relation to hull strength. That is to say, zero degradation until disabled, and then full degradation.

You can't justify your weapons with one hand and dismiss the reason your justification works on the other. Either it works that way or it doesn't.

**Comment by ChamEV** [comment]
> The reason why beam weapons are kind of a niche case is because of the time between the last shot (which usually happens when the exact moment the ship is disabled), and the last shot connecting (beams are instant, where no other weapons are). This means beams always stop at exactly the right moment, and all other weapons are still firing at you.

Perhaps we should add a delay before beam weapons stop firing, half second or so due to human reaction time just to balance this across all weapon types.

**Comment by ghoulavenger** [comment]
@ChamEV That would certainly make beam weapons less dominant in disabling only.

**Comment by Zitchas** [comment]
It seems that I am completely missing your explanation, then. My request is for a `disabling` attribute for weapons which specify that it does only 25% of its normal damage when the ship it is impacting is disabled. This does not require any calculation for any other projectiles, nor does it require some sort of jedi-like "reduce firing when it thinks it'll reach disabled" or anything like that. (That would be more the realm of intelligent decision making for the "disabling" personality.) It is quite simple: When the weapon impacts the ship (which is to say, at the very moment its damage is applied to it, it would be subject to a basic if statement: If the ship is disabled, the damage applied is multiplied by 0.25. If the ship is still functional, it applies damage as normal. That's it. 

The EMP torpedo is an excellent of example of a situation where I think this could work, as it is a weapon that is working via damage being applied to the ships systems. So long as the ship is active (that is to say, has internal control systems that still have power), the weapon is effective at doing damage. When those systems are inactive, an EMP burst is far, far less effective against it. 

As such, in ES's current system, the hull health of a ship is a holistic value: It represents the armor, the ship electronics, control systems, hydraulics, and everything else. For simplicity, all these are rolled into one number. The `disabling` attribute is a way of providing a degree of differentiation in stating that a weapon is actually doing its work by focusing on the systems aspect of the hull value, and that once the internal systems are no longer a useful target (because they are disabled and non-function), that weapon is no longer as effective.

No, I'm not dismissing why it works: I'm specifying that the disabling mechanism works as well as the ship it is attacking is working. Just like you said, the disabling attribute as proposed follows the same degradation curve as ship performance. Which is to say, fully effective until the ship reaches the disabling point, and then a sudden drop to low effectiveness. I can think of a variety of ways this could be. Perhaps ships in ES universally have large numbers of redundant systems, so part of the hull health being whittled away is representing all the redundant systems being taken out one by one, with the ship switching from fully functional to disabled when the last redundancy is taken out. 

It would be unnecessarily confusing to have a weapon that is primarily effective against active ships suffer a gradual degradation over time based on hull health when ships do not suffer the same degradation of performance. It adds unnecessary complexity to the process, and is only realistic in some scenarios, not all. Hence the request for a binary application.

That, however, is a discussion for a suggestion to reform the ship health and structural integrity system of ES, which, frankly, I'm not suggesting here and I'm not terribly interested in trying to include here.

**Just to get this discussion back on track:**
The request / proposal is:
```That a new attribute be added to ES that can be applied to weapons that specifies that this weapon is less effective against ships that are disabled, as represented by the fact that their damage is reduced to 25% of its normal amount when they strike a ship that is disabled. ```

This damage reduction would be applied at the point of impact only. It would not be predictive, it would not try to take into account other potential damage sources that are incoming. It would be a strictly binary application based on whether the recipient of the damage is or is not disabled at that moment. 

Yes, especially for things that deal a lot of damage, it would still have the potential to "overflow" a lot of damage if it hit when the target only has a couple points of hull health left, perhaps even destroying it. This is acceptable, and is not considered to be a problem or something that needs to be resolved.

This request/proposal is *only* for the introduction of this attribute, which would enable a greater variety of weapons and utility scenarios, all of which would need to be added after the attribute is implemented. Discussion on which weapons should be changed (if any) or added (if any) would be the subject of a future discussion, and likely predominantly the domain of plug-ins.

In short, this proposal isn't seeking to change any existing weapon, but merely add an additional option for creating new and interesting possibilities for content creators. Suggestions as to how it might be applied both in vanilla content and plug-ins are merely ideas.

As a side note, this also helps bring ES one step closer to feature parity with the EV series, which is the primary inspiration for this game.

**Comment by petervdmeer** [comment]
The change you are proposing doesn't sound to complex (a couple of lines of code just before applying received damage on a ship), but if we implement this, then instead of a "disabling" attribute, I would prefer a "hull damage on disabled" attribute.

In your example you would specify a "hull damage on disabled" value of one fourth of the original value of the weapon. I can imagine that other plugin builders would like to set such a value to much higher than the original "hull damage" value of the weapon (to finish off disabled ships faster).

If this feature is only used in plugins and not in the main game, then I expect that this feature can still be worthwhile to implement.

If "hull damage on disabled" is not set, then the original "hull damage" would be applied on ships that are hit by a weapon.

Would the "hull damage on disabled" option work for you? And does somebody have a better name for such option?

**Comment by Zitchas** [comment]
I hadn't thought of setting it as a value, but yes, that would work quite nicely.

So, default value would be 1, which would result in the weapon's normal damage, with lower values resulting in less damage being dealt, and higher values being more damage. 

What about negative values? Theoretically, that would result in a weapon that... Heals the target ship? I don't know how the game would handle that, but I think it'd be great to have a weapon that heals disabled ships. I'm not sure what sort of lore would work for that, though.

**Comment by ChamEV** [comment]
Better to have it's own value rather than a multiplier.

`"disabled damage" x` where x is by default the `"hull damage"` value unless specified.

Damage to hull after being disabled would be applied the same way that hull damage is applied after the shield is hit.  That is if 40% of the weapon damage is used to get to disabled, then the remaining 60% of damage would use the `"disabled damage"` value.

**Comment by Zitchas** [comment]
OK, using an actual value instead of a multiplier would work fine.

That being said, I wouldn't put too much effort into the extra complexity of fractional damage. There's a lot of potential explanations for why damage overflows over. If it can manage it, great, but I don't think that is as important as the first part.

**Comment by ChamEV** [comment]
It would keep things consistent as it's already done for any weapon that breaks shields and then does the remainder to the hull.

**Comment by Zitchas** [comment]
Alright then, if the math is already there to handle it, then might as well, I suppose. Sounds like you know your way around the code enough that you might be able to add it in?

**Comment by petervdmeer** [comment]
> What about negative values? Theoretically, that would result in a weapon that... Heals the target ship? I don't know how the game would handle that, but I think it'd be great to have a weapon that heals disabled ships. I'm not sure what sort of lore would work for that, though.

The game already supports negative damage (see #861 ), but the AI cannot handle healing guns at the moment. (The AI still targets enemies with such weapons.)

What I would expect from a weapon that has a positive "hull damage" attribute and a negative "disabled damage" attribute is that it disables the ship, but that any hit after disabling will heal the target ship to just below the disabled boundary with every hit.
I don't want such a weapon to cause oscillations between disabled and non-disabled.

Such weapon could be called a "Stasis Torpedo", "Preservation Beam" or "Suspension Gun". And I guess that in-lore some Xenophobic alien race could use it to disable enemies and investigate them. Or the Wanderers might use such weapon to disable enemies, relocate them and then heal them (assuming they also get a full healing gun).

**Comment by petervdmeer** [comment]
The remnant have a history of plundering other ships, it is also a suitable type of weapon for them.

**Comment by petervdmeer** [comment]
Just to avoid waiting for each-other and also to avoid double work;
@ChamEV Is this something that you planned to implement?

**Comment by ChamEV** [comment]
I was under the impression that you were working on it already.  But if not I would like to take a stab at it.

**Comment by ChamEV** [comment]
Have this basically working now except for 2 things.

* When `disabled damage` is 0, ships don't stay disabled, they get right on the threshold, are displayed disabled for a second and then un-disables itself.
* To set the default value for `disabled damage` to be `hull damage` either `disabled damage` must be specified after `hull damage` which is not really ideal, or I will need to set some constant, probably very big negative number that no one would ever randomly choose.

**Comment by petervdmeer** [comment]
Sounds like good progress. I'm looking forward to your pull request.

**Comment by ChamEV** [comment]
Uploaded an initial version, no PR yet, still contains debug code so please ignore those lines (lines without indent are debug but not all debug lines have no indent).

https://github.com/endless-sky/endless-sky/compare/master...ChamEV:disabledDamageType#diff-3da447d56ddc2a3ec88a8f596978e7d9R2661

There are two nitpicks in this current version.

on line 2652 it may be necessary to change

`hullFraction = max(nonDisabledHull/hullDamage,0.d);`
to
`hullFraction = min(max(nonDisabledHull/hullDamage,0.d) * 1.00001, 1.d);`

which will slightly bias the hull fraction upward and reduce the remaining disabled fraction, but will have no effect if it's already zero.  This will protect against any edge case double rounding errors if a weapon does 0 disabled damage.

There were 2 checks of
`hull < minimumHull` which I changed to `hull <= minimumHull` in ship.cpp lines: 1652 & 2045

The second current issue is just an ugly solution for disabled damage's default value not being 0 makes for some ugly code exceptions in OutfitInfoDisplay.  The only other solution I can reasonably conceive of is storing the default values of each damage type and checking if they're not default.  The default `disabled damage` value changes per weapon, so it would be a non const non static array almost identical to the damage array in weapon.h with the 3rd element being set to the hull damage value.

**Comment by Zitchas** [comment]
Those pictures look good!

**Comment by petervdmeer** [comment]
> both checks are changed from hull < MinimumHull() to hull <= MinimumHull(). A very minor change but necessary.

This (and the other workaround) is not necessary if the disabled-damage itself is applied/calculated from MinimumHull()-1 instead of from MinimumHull().
I expect that nobody would mind if the disabled-damage is applied only from a full point below the MinimumHull, but making a ship already disabled when it is at MinimumHull (instead of below it) is not desired.

Also, if we would heal ships using negative "disabled damage", then healing to a full point below MinimumHull should be sufficient as well (to prevent ships from toggeling between healed and not healed).

For the OutfitInfoDisplay I would like the "disabled damage" to be shown on the same line as "hull damage" if the "disabled damage" is different from the "hull damage".
So:

> hull damage / second:         50  (if disabled damage is the same as hull damage)
> hull damage / second:    50 / 10  (if disabled damage is different)
> hull damage / shot:           25  (if disabled damage is the same as hull damage)
> hull damage / second:     25 / 5  (if disabled damage is different)

**Comment by ChamEV** [comment]
I like the first idea, I went with 0.25 as it's even less impactful but still works, and is a fraction that can be expressed exactly by a floating point.

I was thinking of using the text "damage if disabled: 25" to make it more clear.  Your solution is more compact however, but requires explanation in the outfits description which would reasonably be expected.

I'm 50/50 between either of them.

**Comment by ChamEV** [comment]
Tried a few things.

* `hull damage / second:     50 / 10`
This was confusing from a player perspective, does it do 50 damage every 10 seconds?
* `hull damage / second:      50 (10)`
Still hard to explain the 10 in the space provided, was not liking the tool tip option either and couldn't easily do separate tool tips if it was there or not.
* `hull damage if disabled / second: 10`
On it's own line, it's too long, leaving only 2-3 digits space.
* `hull damage / second:       50`
* `___if disabled / second:       0` ( '_' are spaces)
This is the best one I think, it's clear what it is, and that it applies to the hull.  Has it's own tooltip.

![image](https://user-images.githubusercontent.com/51398924/60476367-3864bc80-9c30-11e9-9e94-c3fb1b0b355a.png)

If we had an extended character set we could use this corner character  `   ┗if disabled / second:   32`

Here's a video of it in action: https://streamable.com/y35rs

**Comment by petervdmeer** [comment]
I also like the option with spaces best. Looks very clear.
